lib/libzutil/os/linux/zutil_device_path_os.c

```diff
@@ -529,17 +529,19 @@ zfs_dev_is_whole_disk(const char *dev_name)
 {
 	struct dk_gpt *label;
 	int fd;
+	int rc;
 
-	if ((fd = open(dev_name, O_RDONLY | O_DIRECT | O_CLOEXEC)) < 0)
+	if ((fd = open(dev_name, O_RDONLY | O_DIRECT | O_CLOEXEC)) < 0) {
 		return (B_FALSE);
+	}
+	(void) close(fd);
```

Comment:
`fd` is expected by `efi_alloc_and_init()`.

Reply:
Good catch!